tests/topotests/ospf_single_switch/test_ospf_single_switch.py

```diff
@@ -0,0 +1,170 @@
+    tgen.start_topology()
+
+    # Starting Routers
+    router_list = tgen.routers()
+
+    for rname, router in router_list.items():
+        logger.info("Loading router %s" % rname)
+        router.load_frr_config(os.path.join(CWD, "{}_frr.conf".format(rname)))
+
+    # Initialize all routers.
+    tgen.start_router()
+
+
+def teardown_module(mod):
+    "Tear-down the test environment"
+    tgen = get_topogen()
+    tgen.stop_topology()
+
+
+def test_all_routes_advertised():
+    tgen = get_topogen()
+
+    if tgen.routers_have_failure():
+        pytest.skip("Skipped because of router(s) failure")
+
+    # networks advertised by each router and the expected next hops
+    networks = {
+        "r1": ("10.0.1.0/24", "203.0.113.1"),
+        "r2": ("10.0.2.0/24", "203.0.113.2"),
+        "r3": ("10.0.3.0/24", "203.0.113.3"),
+        "r4": ("10.0.4.0/24", "203.0.113.4"),
+        "r5": ("10.0.5.0/24", "203.0.113.5"),
+        "r6": ("10.0.6.0/24", "203.0.113.6"),
+        "r7": ("10.0.7.0/24", "203.0.113.7"),
+        "r8": ("10.0.8.0/24", "203.0.113.8"),
+    }
+
+    for router_orig in tgen.routers().keys():
+        for router_dest, network in networks.items():
+            if router_orig != router_dest:
+                input_dict = {
+                    router_orig: {
+                        "static_routes": [
+                            {
+                                "network": network[0],
+                            }
+                        ]
+                    }
+                }
+                result = verify_ospf_rib(tgen, router_orig, input_dict, next_hop=network[1])
+                assert result is True, "Error: {}".format(result)
+                result = verify_rib(tgen, "ipv4", router_orig, input_dict, next_hop=network[1])
+                assert result is True, "Error: {}".format(result)
+
+                check_route(router_orig, network[0], network[1])
+
+
+def check_route(router_name, network, expected_nexthop):
+    """
+    Checks if the given network is present on the given router and has the
+    expected next hop.
+    """
+    tgen = get_topogen()
+    router = tgen.gears[router_name]
+
+    expected_response = {
+        network: [
+            {
+                "prefix": network,
+                "protocol": "ospf",
+                "nexthops": [
+                    {
+                        "ip": expected_nexthop,
+                        "active": True,
+                    },
+                ],
+            },
+        ],
+    }
+
+    test_func = partial(
+        topotest.router_json_cmp,
+        router,
+        f"show ip route {network} json",
+        expected_response,
+    )
+    _, result = topotest.run_and_expect(test_func, None, count=60, wait=1)
+    assert (
+        result is None
+    ), f"{router_name} (ospfd): no route {network} via {expected_nexthop}"
+
+    address = network.split("/")[0]
+    output = router.cmd(f"ip route get {address}")
```

Comment:
Please use `ip -j route ...`

Reply:
I agree that using `ip -j route` is a better option, and it works fine on Debian 11. However, it does not seem to work on all operating systems. For example, it does not work on Ubuntu 18.04.

Reply:
We already use this pattern, and 18.04 is going to be deprecated at all soon (at least for our CI) :)